audits/multicall/202408-threat-model-multicaller.md

```diff
@@ -0,0 +1,77 @@
+4. The `Multi-call Signer` uses an account with `MULTICALL_SIGNER_ROLE` to sign the function calls and returns the signature back to the `Central Authority`. 
+5. The `Central Authority` returns the list of function calls and the signature to the client.
+6. The `Client` approves the `Guarded Multi-caller` contract as a spender for their `ERC721Card` NFT.
+7. The `Client` submits a transaction to the `Guarded Multi-caller` contract to execute the list of function calls.
+8. The `GuardedMulticaller` contract calls the `ERC721Card` contract to burn the NFT and calls the `ERC721Pet` contract to mint a new NFT to the player’s wallet.
+
+# Attack Surfaces
+
+## Compromised Admin Keys
+The compromised admin keys are able to assign the `MULTICALL_SIGNER_ROLE` to malicious parties and allow them to generate signatures that are valid to invoke function calls that are damaging, e.g. minting tokens on ERC20 contracts to attackers' addresses, or burning tokens from wallets that grant the contract approvals to their tokens.
+
+## Compromised Signer Keys
+The compromised signer keys can generate signatures that are valid to invoke function calls that are damaging, e.g. minting tokens on ERC20 contracts to attackers' addresses, or burning tokens from wallets that grant the contract approvals to their tokens.
+
+# Attack Mitigation
+
+- The keys associated with the `DEFAULT_ADMIN_ROLE` and `MULTICALL_SIGNER_ROLE` should be operated by a secure manner, for example a multi-signature wallet such that an attacker would need to compromise multiple signers simultaneously, or a securely stored hardware wallet.
+- If admin keys are compromised, admins of token contracts should revoke `MINTER_ROLE` granted on the Guarded Multi-caller contract. 
+- If signer keys are compromised, admins of Guarded Multi-caller contracts should revoke the signer keys.
+- If users grant access to their tokens, they should only grant approvals to the tokens needed for the multi-call transaction. If users have to grant access to all of their tokens, they should revoke the access immediately after the multi-call transaction is completed. With smart contract wallet's batch transaction, users can batch approvals, multi-call transaction, or approval revocation in one batch transaction.
+
+# Functions
+
+Functions that _change_ state:
+| Name | Function Selector | Access Control |
+| ------------------------------------------------------------- | ----------------- | --------------------- |
+| execute(address,bytes32,(address,string,bytes)[],uint256,bytes) |  | Caller must have a valid signature |
+| grantMulticallSignerRole(address) |  | DEFAULT_ADMIN_ROLE |
+| revokeMulticallSignerRole(address) |  | DEFAULT_ADMIN_ROLE |
+| grantRole(bytes32,address) |  | DEFAULT_ADMIN_ROLE |
+| revokeRole(bytes32,address) |  | DEFAULT_ADMIN_ROLE |
+| renounceRole(bytes32,address) |  | DEFAULT_ADMIN_ROLE |
```

Comment:
Change to be in alphabetical order - it makes it simpler to audit the list of functions